cmake/Fortran.cmake

```diff
@@ -18,34 +18,80 @@ if(APPLE)
 
     enable_language(Fortran)  # Needed to populate ${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES}
+
+    include(CheckFortranCompilerFlag)
+    check_fortran_compiler_flag(-static-libgfortran HAS_STATIC_LIBGFORTRAN)
+    check_fortran_compiler_flag(-static-libgcc HAS_STATIC_LIBGCC)
+    check_fortran_compiler_flag(-static-libquadmath HAS_STATIC_LIBQUADMATH)
+    message(DEBUG "HAS_STATIC_LIBGFORTRAN=${HAS_STATIC_LIBGFORTRAN}, HAS_STATIC_LIBGCC=${HAS_STATIC_LIBGCC}, HAS_STATIC_LIBQUADMATH=${HAS_STATIC_LIBQUADMATH}")
     # Debug:
     # set(Fortran_VERBOSE "-v")
-    # Find the static libquadmath maually
-    find_library(static-libquadmath NAMES quadmath.a libquadmath.a REQUIRED PATHS ${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES})
-    message(STATUS "Found static-libquadmath at ${static-libquadmath} by searching in CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES=${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES}")
+
     target_compile_options(fortran_project_options INTERFACE
       ${Fortran_VERBOSE}
     )
-    target_link_options(fortran_project_options INTERFACE
-      ${Fortran_VERBOSE}
-      -nodefaultlibs
-      -static-libgfortran
-      -lgfortran
-      -static-libgcc
-      -lgcc
-      -lemutls_w
-      ${static-libquadmath}
-      -lSystem
-    )
 
-    message(DEBUG "CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES}")
-    #    => CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-gfortran;emutls_w;gcc;quadmath;emutls_w;gcc;gcc
-    list(REMOVE_DUPLICATES CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES)
-    list(TRANSFORM CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES REPLACE "quadmath" " ${static-libquadmath}")
-    message(DEBUG "CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES-${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES}")
+    if (NOT HAS_STATIC_LIBGFORTRAN OR NOT HAS_STATIC_LIBGCC)
+      message(FATAL_ERROR "Please check your compiler. gfortran supports -static-libgcc and -static-libgfortran since at least 2007, but HAS_STATIC_LIBGFORTRAN=${HAS_STATIC_LIBGFORTRAN}, HAS_STATIC_LIBGCC=${HAS_STATIC_LIBGCC}")
+    endif()
+    if (HAS_STATIC_LIBQUADMATH)
+      target_link_options(fortran_project_options INTERFACE
+        ${Fortran_VERBOSE}
+        -static-libgfortran
+        -static-libgcc
+        -static-libquadmath
+      )
+    else()
+      # Find the static libquadmath manually
+      find_library(static-libquadmath NAMES quadmath.a libquadmath.a REQUIRED PATHS ${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES})
+      message(STATUS "Found static-libquadmath at ${static-libquadmath} by searching in CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES=${CMAKE_Fortran_IMPLICIT_LINK_DIRECTORIES}")
+
+      message(DEBUG "CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES=${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES}")
+      # On brew gcc 12.2.0
+      #    => CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES=gfortran;emutls_w;gcc;quadmath;emutls_w;gcc;gcc
+      # On https://github.com/fxcoudert/gfortran-for-macOS
```

Comment:
I think moving forward we'll just always plan on brewing gcc, but this is good to have documented, thanks.